WordPress/gutenberg
# Make link UI rich previews target agnostic and fix unwanted preview for internal URLs

## Description
This PR has two purposes:

1. To fix [this bug where the rich preview shows in the Link UI for internal URLs ](https://github.com/WordPress/gutenberg/issues/32657)
2. As a consequence of `#1`, slightly refactor the rich data APIs to make them less _external_ URL specific. This has the advantage of fixing `#1` and also preparing the link UI for showing rich previews of _internal_ URLs in the future.

**Note**: changes to test solely down to renaming of the main fetch function.

Closes https://github.com/WordPress/gutenberg/issues/32657

## Questions

~@youknowriad Where do you think the new `fetchRichUrlData` function should reside? I was thinking `@wordpress/core-data`?~

Suggest we ship "as is" and revisit where to move the function in a followup.

## How has this been tested?

#### Automated Testing

I will add unit tests for the `fetchRichUrlData` once we have found a suitable home for it.

#### Manual testing instructions

* Create post with text in it.
* Add x4 links 
    - one to an internal page/post
    - one to a remote URL (eg: `www.wordpress.org`).
    - one to a remote URL which includes the internal (URL of the site) as a query string (eg: `www.wordpress.org?qs=http://localhost:8888`).
    - one to a URL which uses a non-http protocol (eg: `tel:123456`).
* Click on all the links to preview them one by one.
* _Both_ the external links should show the rich preview.
* _Both_ the external links should show a loading state whilst fetching (trying throttling your network connection).
* The _internal_ URL should not show a loading state or any form of "rich" preview.
* The non-http URL should not show a loading state or any form of "rich" preview.
## Screenshots 

https://user-images.githubusercontent.com/444434/121881455-386c0300-cd07-11eb-8ff0-74d119963b33.mp4



## Types of changes





## Checklist:
